File: src/main/java/kr/mjc/rachel/spring/jdbc/user/AddUser.java
package kr.mjc.rachel.spring.jdbc.user;

import kr.mjc.rachel.basics.jdbc.user.User;
import kr.mjc.rachel.basics.jdbc.user.dao.UserDao;
import kr.mjc.rachel.spring.jdbc.AppConfig;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import org.springframework.dao.DataAccessException;

import java.util.Scanner;

@Slf4j
public class AddUser {
  public static void main(String[] args) {
    System.out.print("Input email password name : ");
    User user = new User();
    Scanner scanner = new Scanner(System.in);
    user.setEmail(scanner.next());
    user.setPassword(scanner.next());
    user.setName(scanner.next());
    scanner.close();

    ApplicationContext applicationContext =
        new AnnotationConfigApplicationContext(AppConfig.class);
    UserDao userDao = applicationContext.getBean(UserDao.class);
    try {
      userDao.addUser(user);
      log.info("입력 완료. {}", user);
    } catch (DataAccessException e) {
      log.info("이메일 중복. {}", e.getCause().toString());
    }
  }
}

File: src/main/kotlin/kr/mjc/jacob/basics/jdbc/user/dao/AddUser.kt
package kr.mjc.jacob.basics.jdbc.user.dao

import kr.mjc.jacob.basics.jdbc.user.User
import java.util.*

fun main() {
  print("Get - email password name ? ")
  val user = User().apply {
    Scanner(System.`in`).use {
      email = it.next()
      password = it.next()
      name = it.next()
    }
  }

  UserDaoImpl.addUser(user)
  println(user)
}
